socorro/external/crashstorage_base.py

```diff
@@ -253,368 +242,65 @@ def remove(self, crash_id):
         raise NotImplementedError("remove is not implemented")
 
 
-class PolyStorageError(Exception, MutableSequence):
-    """Exception container holding a sequence of exceptions with tracebacks
-
-    :arg message: an optional over all error message
-
-    """
-
-    def __init__(self, *args):
-        super().__init__(*args)
-        self.exceptions = []  # the collection
-
-    def gather_current_exception(self):
-        """Append the currently active exception to the collection"""
-        self.exceptions.append(sys.exc_info())
-
-    def has_exceptions(self):
-        """Boolean opposite of is_empty"""
-        return bool(self.exceptions)
-
-    def __len__(self):
-        """Length of exception collection"""
-        return len(self.exceptions)
-
-    def __iter__(self):
-        """Start an iterator over the sequence"""
-        return iter(self.exceptions)
-
-    def __contains__(self, value):
-        """Search the sequence for a value and return true if it is present"""
-        return self.exceptions.__contains__(value)
-
-    def __getitem__(self, index):
-        """Get a specific exception by index"""
-        return self.exceptions.__getitem__(index)
-
-    def __setitem__(self, index, value):
-        """Set the value for an index in the sequence"""
-        self.exceptions.__setitem__(index, value)
+class InMemoryCrashStorage(CrashStorageBase):
```

Comment:
I added this to make it easier to write tests, but in retrospect, I probably could have written the tests using the FSCrashStorage. I'm leaving it here for now, but maybe we should remove it later.